pyi.py

```diff
@@ -187,6 +239,76 @@ def in_class(self) -> bool:
         """Determine whether we are inside a `class` statement"""
         return bool(self._class_nesting)
 
+    def _check_typing_object(
+        self, node: ast.Attribute | ast.ImportFrom, module_name: str, object_name: str
+    ) -> None:
+        if object_name in _BAD_COLLECTIONS_ALIASES:
```

Comment:
This logic is a bit hard to follow because we don't always check the module name. You have to look at the other methods to see that this gets called only for typing/typing_extensions.

What if instead we always called _check_typing_object for each attribute and import, and then here had logic like
```
if module_name == "typing":
    if object_name not in _BAD_BUILTINS_ALIASES:
        return
    replacement = ...
elif module_name == "typing_extensions":
     # different logic for deciding if the name is bad
self.error(...)
```
That makes it easier to see at a glance which names will produce errors for each module.

Reply:
Okay!